scenariorun/src/main/kotlin/com/cosmotech/scenariorun/workflow/argo/ArgoWorkflowService.kt

```diff
@@ -374,6 +377,54 @@ internal class ArgoWorkflowService(
     return healthBuilder.withDetail("url", apiClient.basePath).build()
   }
 
+  override fun stopWorkflow(
+      organizationId: String,
+      scenarioRunId: String,
+      workflowName: String
+  ): ScenarioRunStatus {
+    var workflow = Workflow()
+    try {
+      workflow =
+          newServiceApiInstance<WorkflowServiceApi>(this.apiClient)
+              .workflowServiceStopWorkflow(
+                  csmPlatformProperties.argo.workflows.namespace,
+                  workflowName,
+                  WorkflowStopRequest())
+    } catch (e: ApiException) {
+      System.err.println("Exception when calling WorkflowServiceApi#workflowServiceStopWorkflow")
+      System.err.println("Status code: " + e.code)
+      System.err.println("Reason: " + e.responseBody)
+      System.err.println("Response headers: " + e.responseHeaders)
+    }
+
+    return ScenarioRunStatus(
```

Comment:
There is already a function which do this mapping "workflow status" => scenarioRunStatus : getScenarioRunStatus
Refactor both to have only one mapping function if possible.